Fix iconv_mime_decode() while using mbstring extension

cedric-anne opened the pull request:
1. `mb_decode_mimeheader` has only one argument (see https://www.php.net/manual/en/function.mb-decode-mimeheader.php).

Calling it with 3 arguments triggers following error: `mb_decode_mimeheader() expects exactly 1 argument, 3 given in ./vendor/symfony/polyfill-iconv/bootstrap80.php at line 57`.

2. `iconv_mime_decode()` `$encoding` argument is not correctly handled. Indeed, it should be used to convert result in expected encoding.

nicolas-grekas commented:
In eg https://www.tutorialspoint.com/php-how-to-decode-a-mime-header-field-using-iconv-mime-decode-function
the function is described as having 3 args.
Can you please dig into php's history and check in which version this changed?

stof commented:
@nicolas-grekas this is describing the iconv method, not the mb one.

However, ignoring the iconv arguments entirely in the polyfill implementation looks wrong to me. That won't produce a working polyfill, as those arguments are not no-op in PHP.

nicolas-grekas reviewed (approved):
Oh, that's why I was confused. :+1: then.
You're right about the arguments @stof 
We need some that is affected to send a PR thought.

cedric-anne commented:
I added a commit to correctly handle the `encoding` parameter of `iconv_mime_decode()` method, and added some tests. I also updated the issue title and description to fit these changes.

nicolas-grekas commented:
Thank you @cedric-anne.
